pennylane/_qutrit_device.py

```diff
@@ -0,0 +1,390 @@
+
+        Returns:
+            array[float]: list of the probabilities
+        """
+
+        wires = wires or self.wires
+        # convert to a wires object
+        wires = Wires(wires)
+        # translate to wire labels used by device
+        device_wires = self.map_wires(wires)
+
+        sample_slice = Ellipsis if shot_range is None else slice(*shot_range)
+        samples = self._samples[sample_slice, device_wires]
+
+        # convert samples from a list of 0, 1 integers, to base 10 representation
+        powers_of_three = 3 ** np.arange(len(device_wires))[::-1]
+        indices = samples @ powers_of_three
+
+        # count the basis state occurrences, and construct the probability vector
+        if bin_size is not None:
+            bins = len(samples) // bin_size
+
+            indices = indices.reshape((bins, -1))
+            prob = np.zeros([3 ** len(device_wires), bins], dtype=np.float64)
+
+            # count the basis state occurrences, and construct the probability vector
+            for b, idx in enumerate(indices):
+                basis_states, counts = np.unique(idx, return_counts=True)
+                prob[basis_states, b] = counts / bin_size
+
+        else:
+            basis_states, counts = np.unique(indices, return_counts=True)
+            prob = np.zeros([3 ** len(device_wires)], dtype=np.float64)
+            prob[basis_states] = counts / len(samples)
+
+        return self._asarray(prob, dtype=self.R_DTYPE)
+
+    def marginal_prob(self, prob, wires=None):
+        r"""Return the marginal probability of the computational basis
+        states by summing the probabiliites on the non-specified wires.
+
+        If no wires are specified, then all the basis states representable by
+        the device are considered and no marginalization takes place.
+
+        .. note::
+
+            If the provided wires are not in the order as they appear on the device,
+            the returned marginal probabilities take this permutation into account.
+
+            For example, if the addressable wires on this device are ``Wires([0, 1, 2])`` and
+            this function gets passed ``wires=[2, 0]``, then the returned marginal
+            probability vector will take this 'reversal' of the two wires
+            into account:
+
+            .. math::
+
+                \mathbb{P}^{(2, 0)}
+                            = \left[
+                               |00\rangle, |10\rangle, |01\rangle, |11\rangle
+                              \right]
+
+        Args:
+            prob: The probabilities to return the marginal probabilities
+                for
+            wires (Iterable[Number, str], Number, str, Wires): wires to return
+                marginal probabilities for. Wires not provided
+                are traced out of the system.
+
+        Returns:
+            array[float]: array of the resulting marginal probabilities.
+        """
+
```

Comment:
In this function also, it looks like the only changes are a few 2 to 3?

Reply:
Yep. Almost all of the functions that are overriden in `QutritDevice` only include changes to change a few 2's to 3's. One exception is `statistics()`, which has been overriden to not include the `VnEntropy` and `MutualInfo` return types for observables as they aren't currently supported on `QutritDevice`.

Speaking of those measurements, I just realized I should override them for `QutritDevice` as well such that their computation is explicitly not supported until later.